pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -181,12 +179,19 @@ func updateStorage(set *apps.StatefulSet, pod *v1.Pod) {
 	pod.Spec.Volumes = newVolumes
 }
 
+func initIdentity(set *apps.StatefulSet, pod *v1.Pod) {
+	updateIdentity(set, pod)
+	// Set these immutable fields only on initial Pod creation, not updates.
+	// Set hostname/subdomain in preparation for removal of the annotations.
```

Comment:
This line of the comment is not applicable on this branch, since the annotations were already removed.